benchmarks/src/efficient_nonlocking_print.cpp

```diff
@@ -0,0 +1,38 @@
+#include <benchmark/benchmark.h>
+#include <cstdio>
+#include <format>
+#include <print>
+#include <string>
+#include <string_view>
+#include <utility>
+
+namespace {
+    using PrintType = void (*)(FILE*, std::string_view, std::format_args);
+
+    template <PrintType PrintFunction>
+    void BM_vprint(benchmark::State& state) {
+        for (auto _ : state) {
+            PrintFunction(stdout, "Hello cool I am going to print as unicode\n", std::make_format_args());

```

Comment:
This benchmark inherently emits a ton of output to stdout, which interferes with actually seeing the benchmark results. We should add a comment about the Google Benchmark options that allow a file to be logged.